src/link.cpp

```diff
@@ -2713,7 +2714,15 @@ static void construct_linker_job_macho(LinkJob *lj) {
     lj->args.append("-arch");
     lj->args.append(get_darwin_arch_string(g->zig_target));
 
-    if (g->zig_target->glibc_or_darwin_version != nullptr) {
+    if (g->zig_target->glibc_or_darwin_version == nullptr) {
+        // running `zig0`; `-pie` requires versions >= 10.5; select 10.13
+        if (g->zig_target->os == OsMacOSX && g->zig_target->glibc_or_darwin_version == nullptr) {
```

Comment:
Do we also need to treat apple IOS specially? Could we share the below `else` clause?

Reply:
tbh I don't really know the -pie disposition on those platforms and was careful to only add macos explicit behaviours.

Reply:
re: `else` clause, no this behaviour is for when `glibc_or_darwin_version` is null (and running `zig0`), the last part I have in next commit.